discord/ext/commands/bot.py

```diff
@@ -559,9 +559,18 @@ def cogs(self):
     def _remove_module_references(self, name):
         # find all references to the module
         # remove the cogs registered from the module
+        removed_cogs = []
         for cogname, cog in self.__cogs.copy().items():
             if _is_submodule(name, cog.__module__):
-                self.remove_cog(cogname)
+                try:
+                    removed_cogs.append(cog)
+                    self.remove_cog(cogname)
+                except BaseException as exc:
```

Comment:
Catching BaseException will interfere with things this should not interfere with should they happen at the wrong time.

Reply:
After looking into it a bit more in depth, I agree.